Side navigation variant for Discourse markup (#3052)
canonical-web-and-design/vanilla-framework: templates/docs/patterns/navigation.md

View example of the side navigation pattern with icons
</a></div>

#### Raw HTML

> When HTML markup of side navigation links is generated by external service and Vanilla class names cannot be added to the elements a `p-side-navigation--raw-html` modifier can be used.

Reviewer: ```suggestion
When the HTML markup of the side navigation links is generated by an external service and Vanilla class names cannot be added to the elements, use the `p-side-navigation--raw-html` modifier.
```

> In this variant side navigation pattern will style headings (`h2`, `h3`, etc.) and lists (`ul`, `li`) with links or text elements (`a`, `strong`, `span`) consistently with default side navigation component:

Reviewer: ```suggestion
In this variant, the side navigation pattern will style headings (`h2`, `h3`, etc.) and lists (`ul`, `li`) with links or text elements (`a`, `strong`, `span`) consistently with default side navigation component:
```

- headings (`h2`, `h3`, etc.) get the styling of `p-side-navigation__item--title`

> - lists `ul` is styled as `p-side-navigation__list`, and `li` as `p-side-navigation__item`

Reviewer: ```suggestion
-  lists: `ul` tags are styled as `p-side-navigation__list`, and `li` tags as `p-side-navigation__item`
```

Author: @lyubomir-popov If we want to add colon to `lists:` should we do it for all the items in the listing for consistency? (`headings:`, `text elements:`, `links:`)?
